apollo-router/src/spec/query.rs

```diff
@@ -837,6 +836,17 @@ impl Query {
                 }
             }
         }
+        // Detect if root __typename is asked in the original query (the qp doesn't put root __typename in subselections)
+        // cf https://github.com/apollographql/router/issues/1677
+        let operation_kind_if_root_typename = self.operations.get(0).and_then(|op| {
+            op.selection_set
```

Comment:
thois should use the `operation()` method that is used to get either the named operation, or if `operationName` is not present, the first operation in the query